yt-dlp: 2025.06.30 -> 2025.07.21

ddogfoodd opened the pull request:
Changelog: https://github.com/yt-dlp/yt-dlp/releases/tag/2025.07.21
Diff: https://github.com/yt-dlp/yt-dlp/compare/2025.06.30...2025.07.21

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jonhermansen reviewed (approved):
Tested this locally on x86_64-linux.

I archived a YT video and part of a live stream from Twitch. LGTM.

SuperSandro2000 commented:
Nope, that PR didn't contain a bump. Miss remembered that but there is now a merge conflict. 😅

ddogfoodd committed:
yt-dlp: 2025.06.30 -> 2025.07.21

Changelog: https://github.com/yt-dlp/yt-dlp/releases/tag/2025.07.21
Diff: https://github.com/yt-dlp/yt-dlp/compare/2025.06.30...2025.07.21

jonhermansen reviewed (approved):
Re-tested with latest changes on x86_64-linux. Downloaded a YT video and part of a Twitch stream. LGTM.
